src/test/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+set(BINARY Test${CMAKE_PROJECT_NAME})
+
+find_package( Threads )
+
+configure_file(CMakeLists.txt.in googletest-download/CMakeLists.txt)
+execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
+  RESULT_VARIABLE result
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/googletest-download )
+if(result)
+  message(FATAL_ERROR "CMake step for googletest failed: ${result}")
+endif()
+execute_process(COMMAND ${CMAKE_COMMAND} --build .
+  RESULT_VARIABLE result
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/googletest-download )
+if(result)
+  message(FATAL_ERROR "Build step for googletest failed: ${result}")
+endif()
```

Comment:
gtest must be added to dependencies, not like this.

Reply:
What do you suggest? I went with the recommended method from [here](https://github.com/google/googletest/blob/master/googletest/README.md#incorporating-into-an-existing-cmake-project) but would it be preferable to copy the source to the lib folder?
I found Kodi uses [this](https://github.com/xbmc/xbmc/blob/master/cmake/modules/FindGtest.cmake), am I better off trying to make use of that?

Reply:
There is a system built into add-ons which allows building dependencies. If you look at ffmpegdirect you will see all the dependencies in the `depends` folder. 

I think adaptive has it for expat.

Reply:
as said, you have to add it to depends: https://github.com/xbmc/inputstream.adaptive/tree/Matrix/depends

Reply:
This comment appears to be still open.  Just tried this on Linux, and cmake completely misses downloading googletest on my system.

Reply:
@smf007 yes I'm working on changing to the same method that all the other depends use. I'll have more time over the next few days to finish but I think I'm 90% there